vortex-serde/src/layouts/write/writer.rs

```diff
@@ -141,10 +142,14 @@ impl<W: VortexWrite> LayoutWriter<W> {
                 Validity::NonNullable,
             )?;
 
-            let metadata_table_begin = self.msgs.tell();
+            let dtype_begin = self.msgs.tell();
             self.msgs.write_dtype(metadata_array.dtype()).await?;
+            let dtype_end = self.msgs.tell();
             self.msgs.write_batch(metadata_array.into_array()).await?;
-            chunks.push_front(Layout::flat(metadata_table_begin, self.msgs.tell()));
+            chunks.push_front(Layout::inlined_schema(
+                vec![Layout::flat(dtype_end, self.msgs.tell())],
+                ByteRange::new(dtype_begin, dtype_end),
+            ));
             column_layouts.push(Layout::chunked(chunks.into(), true));
```

Comment:
ok, what's gross here is that the `metadata` chunk is the dtype instead of the stats table, which we only know because of the context (i.e., that it's a "metadata array"). taken out of that context, it's impossible to know which metadata-first-chunk path to take